app/values/resource_identifier/solr_repository.rb

```diff
@@ -0,0 +1,62 @@
+class ResourceIdentifier
+  ##
+  # Repository for querying solr for a single or multiple IDs and getting back a
+  # document object which can return its dependants and timestamps.
+  class SolrRepository
+    class << self
+      # @param [Array<Integer> OR Integer] id ID or array of IDs of records to
+      #   query for.
+      # @return [Record, #timestamp, #dependant_ids] Solr-record powered record
+      #   which can return its timestamp and dependant ids.
+      def find(id)
+        id = Array.wrap(id)
+        return CompositeRecord.new({}) if id.blank?
+        records = solr.query("id:(#{id.join(' OR ')})").map do |r|
```

Comment:
This seems a little heavy to me -- if there are a large number of FileSets, then loading them all, even from Solr, seems like a lot of data to move around to find the most recent timestamp.  Can't we do a Solr query like "id:X OR generic_work_ids_ssim:X", sort by timestamp, and get the timestamp from the first record?